File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/app/AppModule.java
package com.elveselimoski.starterapp.di.app;

import android.app.Application;
import android.content.Context;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class AppModule {

    private Application app;

    public AppModule(Application app) {
        this.app = app;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return app.getApplicationContext();
    }
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/app/AppModule.kt
package com.elveselimoski.starterapp.di.app

import android.content.Context
import com.elveselimoski.starterapp.StarterApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val app: StarterApp) {

    @Provides
    @Singleton
    internal fun provideContext(): Context {
        return app.applicationContext
    }
}
